services/movie_session.py

```diff
@@ -1,43 +1,46 @@
 from django.db.models import QuerySet
+from typing import Optional
+
 
 from db.models import MovieSession, Movie, CinemaHall
 
 
 def create_movie_session(movie_show_time: str,
                          movie_id: int,
                          cinema_hall_id: int) -> None:
-    movie = Movie.objects.get(id=movie_id)
-    hall = CinemaHall.objects.get(id=cinema_hall_id)
+    movie = Movie.objects.get(pk=movie_id)
+    hall = CinemaHall.objects.get(pk=cinema_hall_id)
     MovieSession.objects.create(show_time=movie_show_time,
                                 cinema_hall=hall,
                                 movie=movie)
 
 
-def get_movies_sessions(session_date: str = None) -> QuerySet:
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
     result = MovieSession.objects.all()
     if session_date:
         result = result.filter(show_time__date=session_date)
     return result
 
 
 def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
-    return MovieSession.objects.get(id=movie_session_id)
+    return MovieSession.objects.get(pk=movie_session_id)
 
 
 def update_movie_session(session_id: int,
-                         show_time: str = None,
-                         movie_id: int = None,
-                         cinema_hall_id: int = None) -> None:
-    session = MovieSession.objects.get(id=session_id)
+                         show_time: Optional[str] = None,
+                         movie_id: Optional[int] = None,
+                         cinema_hall_id: Optional[int] = None) -> None:
+    if any([show_time, movie_id, cinema_hall_id]):
+        session = MovieSession.objects.get(pk=session_id)
     if show_time:
         session.show_time = show_time
     if movie_id:
-        session.movie = Movie.objects.get(id=movie_id)
+        session.movie = Movie.objects.get(pk=movie_id)
     if cinema_hall_id:
-        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+        session.cinema_hall = CinemaHall.objects.get(pk=cinema_hall_id)
```

Comment:
you can avoid making additional 3 sql queries setting session.**movie_id** for example